File: core/src/main/java/org/explang/truffle/RuntimeTypeError.java
package org.explang.truffle;

/**
 * Indicates a type error (that should have been prevented at compile time).
 */
public final class RuntimeTypeError extends RuntimeException {
  public RuntimeTypeError(String message) { super(message); }
  public RuntimeTypeError(Throwable cause) { super(cause); }
  public RuntimeTypeError(String message, Throwable cause) { super(message, cause); }
}

File: core/src/main/kotlin/org/explang/syntax/RuntimeTypeError.kt
package org.explang.syntax

/**
 * Indicates a type error (that should have been prevented at compile time).
 */
class RuntimeTypeError : RuntimeException {
  constructor(message: String) : super(message) {}
  constructor(cause: Throwable) : super(cause) {}
  constructor(message: String, cause: Throwable) : super(message, cause) {}
}
